searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp

```diff
@@ -506,6 +510,57 @@ Test::testEndGuard(bool invert)
     EXPECT_FALSE(m.seek(_bvs[0]->size()+987));
 }
 
+class Verifier : public search::test::SearchIteratorVerifier {
+public:
+    Verifier(size_t numBv, bool is_and);
+    ~Verifier();
+
+    SearchIterator::UP create(bool strict) const override;
+
+private:
+    bool _is_and;
+    mutable TermFieldMatchData _tfmd;
+    std::vector<BitVector::UP> _bvs;
+};
+
+Verifier::Verifier(size_t numBv, bool is_and)
+    : _is_and(is_and),
+      _bvs()
+{
+    for (size_t i(0); i < numBv; i++) {
+        _bvs.push_back(BitVector::create(getDocIdLimit()));
+    }
+    for (uint32_t docId: getExpectedDocIds()) {
+        if (_is_and) {
+            for (auto & bv : _bvs) {
+                bv->setBit(docId);
+            }
+        } else {
+            _bvs[docId%_bvs.size()]->setBit(docId);
+        }
+    }
+}
+Verifier::~Verifier() = default;
+
+SearchIterator::UP
+Verifier::create(bool strict) const {
+    MultiSearch::Children bvs;
+    for (const auto & bv : _bvs) {
+        bvs.push_back(BitVectorIterator::create(bv.get(), getDocIdLimit(), _tfmd, strict, false).release());
```

Comment:
in the blueprint code the 'strict' parameter is inherited differently for AND(first child) and OR(all children). This is minor and should not impact the actual testing.

Reply:
Correct. So add a test that strictness is as expected.